api/backend/blueflood/blueflood.go

```diff
@@ -168,16 +169,32 @@ func (b *Blueflood) fetchSingleSeries(metric api.GraphiteMetric, sampleMethod ap
 	}
 
 	// Construct a Timeseries from the result
-	series := make([]float64, len(result.Values))
-	for i, metricPoint := range result.Values {
-		series[i] = reflect.ValueOf(metricPoint).FieldByName(selectResultField).Float()
+
+	values := make([]float64, timerange.Slots())
+	// The array is initialized to NaN.
+	for i := range values {
+		values[i] = math.NaN()
+	}
+
+	for _, metricPoint := range result.Values {
+		// TODO: check the result of .FieldByName against nil (or a panic will occur)
+		value := reflect.ValueOf(metricPoint).FieldByName(selectResultField).Float()
+		// TODO: check the result of .FieldByName against nil (or a panic will occur)
+		timestamp := reflect.ValueOf(metricPoint).FieldByName("Timestamp").Int()
```

Comment:
should just do `.Timestamp` - the only reason why we're doing reflection above is because the above can retrieve one of the many values.